frontends/p4/moveConstructors.cpp

```diff
@@ -136,18 +143,17 @@ class MoveConstructorsImpl : public Transform {
 
     const IR::Node *postorder(IR::ConstructorCallExpression *expression) override {
         if (convert == Region::Outside) return expression;
-        auto tmpvar = refMap->newName("tmp");
+        auto tmpvar = nameGen.newName("tmp");
         auto tmpref = new IR::PathExpression(IR::ID(expression->srcInfo, tmpvar));
         cmap.add(expression, tmpvar);
         return tmpref;
     }
 };
 }  // namespace
 
-MoveConstructors::MoveConstructors(ReferenceMap *refMap) {
+MoveConstructors::MoveConstructors() {
```

Comment:
DoMoveConstructors could now be merged into MoveConstructors since it's only invoking one pass.

Reply:
Yes. I decided to keep things as-is as sometimes these "DoFoo" passes as executed standalone elsewhere. I'm fine you get rid of extra PassManager wrappers where possible if this is ok for downstream.

Reply:
This one is safe for the Tofino code. Can't speak for any other downstream projects.

Reply:
Thanks! I explicitly decided not to touch midend passes for now as these would impact downstream projects more. But in the majority of cases the changes are just removing of `refmap` argument and making ctors explicit if necessary. This might uncover few subtle cases though, e.g. then refmap was stale before some downstream pass, this needs to be fixed anyway :)